### DISCUSSION TITLE
unable to run codespace due to Failed to start VS Code Remote server

### DISCUSSION THREAD

**Discussion by moshederri** [community/community#38410] [Codespaces] [answered]
Hey, 

we are constantly getting the following error when launching codespace:  

An unexpected error occurred that requires a reload of this page.
The workbench failed to connect to the server (Error: )

And after reload, it still doesn't work and there is not indication of a downtime in the [status page](https://www.githubstatus.com/)

Any ideas what could be the problem?

**Comment by david-engelmann** [comment]
Do you have a lot of browser extensions installed?

**Reply by moshederri** [reply]
No, i dont.

Some additional context i gathered:
Most of the time the error returns empty (Error: ) and sometimes (Error: Failed to start VS Code Remote server.).

Looking at the console output while the codespace runs, i get:

```
vscs:web-client:efeb:codespaces-component:connection-hub:liveshare:connection:9895F2E1A03F7DE818C222ED008555FA1083:liveshare:WorkspaceClient Creating proxy for sourceEvent service
async.ts:30 Uncaught (in promise) Canceled: Canceled
...
```

however, few lines after, it seems like codespace is available:
`vscs:web-client:efeb:codespace-state-watcher codespace state -> Available`
 
Then, im getting the following Errors:
```
ERR Error while registering log channel vscode-log:/20221106T235607/remoteTunnel.log Timed out while waiting for file to be created
log.ts:324   
ERR Error while registering log channel file:///20221106T235607/ptyhost.log Timed out while waiting for file to be created

vscs:web-client:efeb:codespaces-component:connection-hub:liveshare:connection:........:liveshare:WorkspaceClient > IVSCodeServerHostService.startRemoteServerResult() error: 
vscs:web-client:efeb:codespaces-component:connection-hub:vscode:connection:.......:vscode Failed to complete start-remote-server with Live Share RPC Failed to start VS Code Remote server. code:-32098
Error: Failed to start VS Code Remote server.

```

the extensions we are using in the devcontainer are:
	"extensions": [
		"ms-python.python",
		"ms-python.vscode-pylance",
		"MarkLarah.pre-commit-vscode",
		"mhutchie.git-graph",
		"PKief.material-icon-theme",
		"sgoley.lookml-syntax-highlighter",
		"ms-vscode.test-adapter-converter"
	]

Thanks for all the helpers!

**Comment by moshederri** [comment]
Some more context:
* we using an image which is based on `mcr.microsoft.com/vscode/devcontainers/python:0-3.9`
* we install the following package - debugpy pre-commit tox poetry dbt-core dbt-spark dbt-spark[PyHive] dbt-spark[ODBC]
* the codespace started failing since yesterday, but worked well before. No changes were made to .devcontainer or our image 
* When running locally, we are getting
```
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient V] < streamManager.getStream()
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient V] > streamManager.getStream() succeeded t:89.0
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient I] openStreamingChannel -> streamId:ServerSharing_1
[Client:extension:1d29:resolver:tunnel:16635:127.0.0.1:liveshare-tunnel[16635]:local-port-forwarder[16635] I] Forwarding connection on channel #2.
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient V] > IVSCodeServerHostService.devContainerConfigChanged: 
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient V] > IVSCodeServerHostService.devContainerConfigChanged: 
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient V] > IVSCodeServerHostService.devContainerConfigChanged: 
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient V] > IVSCodeServerHostService.devContainerConfigChanged: 
[Client:extension:1d29:resolver:resolve:1:connection:....-34b44a1fa326_stable_129500ee4c8ab7263461ffe327268ba56b9f210d:vscode E] Failed to complete start-remote-server with gRPC, attempting to use Live Share RPC> object #0: "Exception was thrown by handler. Exception: Failed to start VS Code Remote server.
Error: Exception was thrown by handler. Exception: Failed to start VS Code Remote server.
    at Object.onEnd (/Users/..../.vscode/extensions/grpc/dist/src/clients/GrpcClientBase.js:29:1)
    at /Users/..../.vscode/extensions/node_modules/@improbable-eng/grpc-web/dist/grpc-web-client.js:1:24231
    at /Users/..../.vscode/extensions/node_modules/@improbable-eng/grpc-web/dist/grpc-web-client.js:1:11490
    at Array.forEach (<anonymous>)
    at e.rawOnError (/Users/..../.vscode/extensions/node_modules/@improbable-eng/grpc-web/dist/grpc-web-client.js:1:11322)
    at e.onTransportHeaders (/Users/..../.vscode/extensions/node_modules/@improbable-eng/grpc-web/dist/grpc-web-client.js:1:9020)
    at NodeHttp.responseCallback (/Users/..../.vscode/extensions/node_modules/@improbable-eng/grpc-web-node-http-transport/lib/index.js:31:1)
    at Object.onceWrapper (node:events:646:26)
    at ClientRequest.emit (node:events:526:28)
    at HTTPParser.parserOnIncomingClient (node:_http_client:618:27)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:128:17)
    at Socket.socketOnData (node:_http_client:482:22)
    at Socket.emit (node:events:526:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)"
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient V] < IVSCodeServerHostService.startRemoteServerResult()
[Client:extension:1d29:resolver:tunnel:16635:127.0.0.1:liveshare-tunnel[16635]:local-port-forwarder[16635] I] Port-forward channel #2 closed.
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient V] > IVSCodeServerHostService.devContainerConfigChanged: 
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient V] > IVSCodeServerHostService.devContainerConfigChanged: 
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient V] > IVSCodeServerHostService.devContainerConfigChanged: 
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient V] > IVSCodeServerHostService.devContainerConfigChanged: 
[Client:extension:1d29:liveshare:connection:....:liveshare:WorkspaceClient E] > IVSCodeServerHostService.startRemoteServerResult() error: 
[Client:extension:1d29:resolver:resolve:1:connection:....-34b44a1fa326_stable_129500ee4c8ab7263461ffe327268ba56b9f210d:vscode E] Failed to complete start-remote-server with Live Share RPC> object #0: "Failed to start VS Code Remote server. code:-32098
Error: Failed to start VS Code Remote server.
    at handleResponse (/Users/..../.vscode/extensions/node_modules/vscode-jsonrpc/lib/main.js:436:1)
    at processMessageQueue (/Users/..../.vscode/extensions/node_modules/vscode-jsonrpc/lib/main.js:263:1)
    at Immediate.<anonymous> (/Users/..../.vscode/extensions/node_modules/vscode-jsonrpc/lib/main.js:247:1)
    at processImmediate (node:internal/timers:466:21)"
[Client:extension:1d29:resolver:resolve:1:connection:....-34b44a1fa326_stable_129500ee4c8ab7263461ffe327268ba56b9f210d:vscode I] Failed to execute start-remote-server: undefined
[perf V] name:start vscode server duration:22308.085333824158
[Client:extension:1d29:resolver:resolve:1:connection:....-34b44a1fa326_stable_129500ee4c8ab7263461ffe327268ba56b9f210d:vscode E] connection failed: > object #0: Failed to start VS Code Remote server. code:-32098
  at handleResponse (/Users/..../.vscode/extensions/node_modules/vscode-jsonrpc/lib/main.js:436:1)
  at processMessageQueue (/Users/..../.vscode/extensions/node_modules/vscode-jsonrpc/lib/main.js:263:1)
  at Immediate.<anonymous> (/Users/..../.vscode/extensions/node_modules/vscode-jsonrpc/lib/main.js:247:1)
  at processImmediate (node:internal/timers:466:21)
[Client:extension:1d29:resolver:resolve:1:connection:....-34b44a1fa326_stable_129500ee4c8ab7263461ffe327268ba56b9f210d:vscode I] connection status changed: connecting -> disconnected, error:> object #0: Failed to start VS Code Remote server. code:-32098
  at handleResponse (/Users/..../.vscode/extensions/node_modules/vscode-jsonrpc/lib/main.js:436:1)
  at processMessageQueue (/Users/..../.vscode/extensions/node_modules/vscode-jsonrpc/lib/main.js:263:1)
  at Immediate.<anonymous> (/Users/..../.vscode/extensions/node_modules/vscode-jsonrpc/lib/main.js:247:1)
  at processImmediate (node:internal/timers:466:21)
[Client:extension:1d29:resolver:resolve:1:connection:....-34b44a1fa326_stable_129500ee4c8ab7263461ffe327268ba56b9f210d:vscode I] firing change event: > object #0: {
  "status": "disconnected",
  "previousStatus": "connecting"
}
[Client:extension:1d29:vscode:onConnectionChanged:....-34b44a1fa326_stable_129500ee4c8ab7263461ffe327268ba56b9f210d I] connecting -> disconnected

```

Any help would be appreciated :)

**Reply by jkeech** [reply]
A couple questions/things to try:
1) Does it work on a new codespace or if you rebuild your current codespace? You could try rebuilding through the GH CLI `gh codespace rebuild`.
2) Can you share more of your devcontainer.json? In particular, what `remoteUser` are you using?

**Reply by jkeech** [reply]
I think I understand what is going on here. The [Python image](https://github.com/devcontainers/images/blob/main/src/python/.devcontainer/devcontainer.json#LL57) has `remoteUser` set to `vscode` by default, and the devcontainer CLI is now [honoring config set in the underlying image](https://github.com/devcontainers/spec/issues/18) if it is present. My guess is that something in your custom Dockerfile is trying to use the `root` user, such as hardcoding `$HOME` to `/root` or something similar, but your devcontainer config likely doesn't have `remoteUser` set to `root`. So in the end, the VS Code server is trying to start using the `vscode` user in the container, but it is pointing to the `/root` home directory, which it doesn't have permissions to install the remote server there.

If you want the devcontainer to run as root, could you try adding `"remoteUser": "root"` to your devcontainer config to override what is in the base image? Or alternatively, to run as `vscode`, remove any config in the Dockerfile which is forcing things back to paths owned by `root`.

**Reply by moshederri** [reply, marked as answer]
hey,

In our Dockerfile we had ENV HOME pointing to `/root` (without overriding `remoteUser`). We didn't understand why after we removed it, codespaces started working again, and after reading your comment, it all makes sense now.

thank you @jkeech !

**Reply by jkeech** [reply]
Glad it's working!

**Reply by kmp9928** [reply]
@moshederri I am having the same error, but can't launch the Codespace so I cannot remove Dockerfiles or anything to try to solve it. How did you do?

**Comment by david-engelmann** [comment]
@moshederri Are you currently hosting it on docker? If you are, it seems like it could be related to docker/wsl being down. What are the results from running the `wsl -l -v` command?

**Comment by huabin** [comment]
Same issue now, I tried to open it on firefox, chrome and safari, and I get the same issue.

<img width="505" alt="error" src="https://github.com/user-attachments/assets/1c0ffacf-16d8-4f98-9925-a4e88f17aa63">

**Reply by huabin** [reply]
BTW, I tried to delete the old codespace and create a new one. Same issue.

**Reply by ShawnSack** [reply]
For a moment, I thought I was the only one.

**Reply by kmendoza197** [reply]
i am getting the same thing

**Reply by eleCollins** [reply]
Same thing!

**Reply by blackopsinc** [reply]
Same Same Its probably production or node is down/broken AWS/GCP down?

**Reply by bartowen** [reply]
same thing

**Reply by joji-harada** [reply]
Yeah, I made a brand new codespace and I'm getting the same thing

**Reply by Dawood-Aamir** [reply]
same error.
<img width="427" alt="Screenshot 2024-07-19 054545" src="https://github.com/user-attachments/assets/0a4f54b4-01b7-4a4d-936b-e3aeaf9af742">

**Reply by blackopsinc** [reply]
Yeah GH status page looks like some s**t is going down https://www.githubstatus.com/

**Reply by xiaoju111a** [reply]
Same it, I afraid lost my code, I forget backend my code 
F*

**Reply by Frtmnml** [reply]
Same Issue here. Seems like something is happening. https://www.githubstatus.com/

**Reply by PhantomLel** [reply]
same

**Reply by Dimension33** [reply]
same.....

**Reply by genericness** [reply]
same

**Reply by Wk6-1** [reply]
same here i deleted my cookies i thought it's something had to do with cookies, but it's not, i'm just hoping that my files did not get deleted

**Reply by Shreneel** [reply]
same

**Reply by severd01** [reply]
same, was using earlier today now broken

**Reply by jumpmanjay** [reply]
+1

**Reply by AlainDevs** [reply]
same, we can't use it for now

**Reply by Vibhor7-7** [reply]
Can use it either

**Reply by blackopsinc** [reply]
After a quick dive in the ocean. Azure is down (VS Code Error) if you have a VPN jump to East Coast and its working. Note you will have to recreate space.

**Comment by JamesHutchison** [comment]
I am presently seeing this issue. I have a codespace and a backup on opposite regions. First codespace worked, then couldn't start it again. Switched to other, worked shut it down, then couldn't start it again.

**Comment by PuppiesUnite421** [comment]
I am having this issue while trying to log into https://cs50.dev/. I get the error message: An unexpected error occurred that requires a reload of this page. The workbench failed to connect to the server (Error: exception was thrown by handler. exception: failed to start vs code remote server.)

**Reply by MinhQuan805** [reply]
same issue

**Reply by supriyaritgithub** [reply]
Delete the current  codespace and create a new one with repository cs50/... it worked for me.

**Reply by SkTauseef** [reply]
what about the files?

**Reply by supriyaritgithub** [reply]
files will be there since it was already auto saved in that (cs50's) repository.

**Reply by neverada** [reply]
can you please explain it step by step cuz i am having the same issue and i have zero idea  what to do

**Reply by supriyaritgithub** [reply]
first, i deleted the codespace  and created a new one in the repository called cs50/... , and then the new codespace opened and asked to rebuild i clicked on and it took some while to load then it worked.

**Comment by Nolikegreek** [comment]
I am having the exact same issue. I found this webpage to look at the status of cs50.dev. https://cs50.statuspage.io/

**Reply by owen800q** [reply]
Same issue

**Reply by yodakkak** [reply]
I can confirm. I've got the same issue, tried 3 different browsers, reset firewall settings, reset browser history and settings and issue persists. Site appears to be down

**Reply by Nolikegreek** [reply]
Seems to work again now

**Comment by yaswant2403** [comment]
For a temporary work-around, opening the codespace in **JupyterLab (Beta)** seems to be working for me.

Seems to be working for me as of 7/18/24 9:51 PM CDT

**Comment by sridbulli** [comment]
An unexpected error occurred that requires a reload of this page.
The workbench failed to connect to the server (Error: exception was thrown by handler. exception: failed to start vs code remote server.)

today plz help

**Reply by moeedcaesar** [reply]
did your problem resolved?

**Comment by chanedwin** [comment]
same issue for me - codespaces seems down

**Comment by neprasad-microsoft** [comment]
I am facing this issue today. Is there a solution for this?

**Reply by achasik** [reply]
same issue for me too

**Reply by moeedcaesar** [reply]
I am also facing this issue, is there any solution around?

**Comment by bandana-edtech** [comment]
I am facing this issue today. Is there a solution for this?

**Reply by RefreX-0** [reply]
same here, it was working fine yesterday

**Reply by bandana-edtech** [reply]
working now

**Reply by neverada** [reply]
hey, u said its working now. I am facing the same issue. It just stopped working today. Just wanted to know what did u do to make it work again

**Reply by jaroslav25** [reply]
working. It usually takes a day or two, or a few hours and starts to work on its own.

**Comment by s-abdulrehman12** [comment]
No

**Comment by s-abdulrehman12** [comment]
I am searching on it

**Reply by moeedcaesar** [reply]
share with us if you find any solution around it

**Reply by RefreX-0** [reply]
yes, please

**Reply by NihalJain** [reply]
following..

**Reply by RefreX-0** [reply]
The codespace is working now for me. They probably fixed it

**Reply by NihalJain** [reply]
> The codespace is working now for me. They probably fixed it

Yes, it's working for me as well now.

**Reply by KazanderDad** [reply]
Not working for me since a few hours. Still not working.

**Reply by jay-kotadiya-incrible** [reply]
Not working for me since a few hours. Still not working.

**Reply by Github-Sri** [reply]
Same here not working for me too.

**Reply by bandana-edtech** [reply]
same here not working again

**Reply by tasneim-github** [reply]
I got the same issue. Do I just wait until they fix it or is there something I should do?

**Reply by Anay-v2** [reply]
Same issue

**Reply by jaroslav25** [reply]
same here

**Reply by Magnesium0306** [reply]
and the same here

**Comment by neverada** [comment]
Thanks for the help, I was actually quite worried on what to do or what not

**Comment by sellwawy** [comment]
It's internet connection problem. try to restart your router, that works for me.

**Comment by florianm** [comment]
It is the year 2025. The world is on fire and so is Codespaces ...again. The error "failed to start VS Code remote server" haunts our developers. Two weeks ago, this error appeared, apparently with no changes to our devcontainer setup. The error persists across repositories, GH organisations, even a blank devcontainer fails to start. It happens across browsers, work locations (home on private network but still corporate firewall/Windows group policy, VPN, work intranet, phone hotspot), browser extensions and privacy blocks (all off).

No solutions found on the web worked. It felt like a networking problem, but no logs and no debugging gave us any actionable insight.

We ended up figuring out that the reason for our breakage was ...our corporate firewall blocking dodgy domains used by "beta" host images.

The solution was:

- GitHub profile > Settings > Codespaces at https://github.com/settings/codespaces
- Host image version preference: Stable

A runner-up would have been "VS Code settings sync" for the possibility of the beta image using a newer VS Code version which might incompatible with the older settings.

(The second problem we experience is that a Codespace, once stopped and re-started, loses its DNS settings and cannot reach GitHub.)

**Reply by partglm** [reply]
I tried but it’s not working

**Reply by florianm** [reply]
Secondary problem: Using a docker compose setup to run a postgres sidecar I tried to customise the postgres container command.

This broke the docker compose build, which broke the Codespace's DNS, which broke the connection of Codespaces to GitHub so I couldn't run any git commands. Nothing anywhere in the logs pointed to the compose file error.

**Comment by MichaelChirico** [comment]
We had `WORKDIR /root` in our Dockerfile in order for a following `COPY` command to WAI, but neither removing it nor adding `"remoteUser": "root"` to the config changed anything -- the codespace still fails to launch with a useless internal error. Frustrating.
